src/Database/Connection.php

```diff
@@ -92,6 +93,14 @@ class Connection implements ConnectionInterface
     protected $_schemaCollection = null;
 
     /**
+     * NestedTransactionRollbackException object instance, will be stored if
+     * the rollback method is called in some nested transaction.
+     *
+     * @var \Cake\Database\Exception\NestedTransactionRollbackException|null
+     */
+    protected $_nestedTransactionRollbackException = null;
```

Comment:
Is there any risk of leaving this property in an inconsistent state? For example throwing an exception before he property is cleared or something like that